cmd/podman/create.go

```diff
@@ -932,3 +953,51 @@ func createContainerFromCreateConfig(r *libpod.Runtime, createConfig *cc.CreateC
 	}
 	return ctr, nil
 }
+
+func makeHealthCheckFromCli(c *cliconfig.PodmanCommand) (*manifest.Schema2HealthConfig, error) {
+	inCommand := c.String("healthcheck-command")
+	inInterval := c.String("healthcheck-interval")
+	inRetries := c.Uint("healthcheck-retries")
+	inTimeout := c.String("healthcheck-timeout")
+	inStartPeriod := c.String("healthcheck-start-period")
+
+	// Every healthcheck requires a command
+	if !c.Flags().Changed("healthcheck-command") {
+		return nil, errors.New("Must define a healthcheck command for all healthchecks")
+	}
+
+	hc := manifest.Schema2HealthConfig{
+		Test: strings.Split(inCommand, " "),
+	}
+	if c.Flags().Changed("healthcheck-interval") {
+		// TODO we should ack a minimum value and check against it
+		intervalDuration, err := time.ParseDuration(inInterval)
+		if err != nil {
+			return nil, errors.Wrapf(err, "invalid healthcheck-internval %s ", inInterval)
+		}
+		hc.Interval = intervalDuration
+	}
+	if c.Flags().Changed("healthcheck-retries") {
+		if inRetries < 1 {
+			return nil, errors.New("healthcheck-interval must be greater than 0.")
```

Comment:
Error message does not match the check.  Should be healthcheck-retries.
Why can't I specify 0?  Ie try it once and then if fail, set unhealthy?

Reply:
i want to defer this to phase three where all the timings will be added.